src/config/index.ts

```diff
@@ -87,11 +87,23 @@ export async function get<T extends SchemaTypeKey>(
 		// Cast the promise as something that produces an unknown, and this means that
 		// we can validate the output of the function as well, ensuring that the type matches
 		const promiseValue = FnSchema.fnSchema[fnKey]();
-		return promiseValue.then((value: unknown) => {
-			const decoded = schemaTypes[key].type.decode(value);
+		return promiseValue
+			.then((value: unknown) => {
+				const decoded = schemaTypes[key].type.decode(value);
 
-			return checkValueDecode(decoded, key, value) && decoded.right;
-		});
+				return checkValueDecode(decoded, key, value) && decoded.right;
+			})
+			.catch(() => {
+				const defaultValue = schemaTypes[key].default;
+				if (defaultValue instanceof t.Type) {
+					const maybeDecoded = (defaultValue as t.Type<any>).decode(undefined);
+
+					return (
+						checkValueDecode(maybeDecoded, key, undefined) && maybeDecoded.right
+					);
+				}
+				return defaultValue as SchemaReturn<T>;
+			});
```

Comment:
This change makes me kind of nervous, I don't think I understand the schema types entirely yet. 

The reason for this is to allow functions to throw. In the case of `deviceArch` and `deviceType`, this allows to memoize only if there was no error.

Does this make sense @Page-, @20k-ultra ?

Reply:
I researched a bit more and I'm fairly confident that this is the correct implementation. In the rare case that a function throws, the catch will try to use the default value. Most functions seem to use t.never, which should lead to the check failing and another exception to be thrown. In the case of deviceType and deviceArch, the default value is the string `'unknown'` which is what should be returned. Since the function throws an error, the result will not be memoized.